### PR TITLE
fix: [PL-24555]: Fix NullPointer in ACL Check API

### PR DESCRIPTION
**Pull request by gupta-utkarsh**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32217)

### CODE REVIEW

**Comment by gupta-utkarsh** [comment]
trigger messagecheck

**Review by reetika04 — APPROVED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32217)* status: 2 file(s) to review. 1 of 2 approvals obtained.

File: access-control/service/src/main/java/io/harness/accesscontrol/acl/api/ACLResourceImpl.java [hunk 1]
```diff
@@ -75,7 +76,10 @@ public ResponseDTO<AccessCheckResponseDTO> get(AccessCheckRequestDTO dto) {
     io.harness.security.dto.Principal contextPrincipal = SecurityContextBuilder.getPrincipal();
     List<PermissionCheckDTO> permissionChecksDTOs = dto.getPermissions();
     Principal principalToCheckPermissionsFor = dto.getPrincipal();
-
+    if (isEmpty(permissionChecksDTOs)) {
+      return ResponseDTO.newResponse(
+          AccessCheckResponseDTO.builder().principal(principalToCheckPermissionsFor).build());
```
Comment (manish-manchanda) [inline access-control/service/src/main/java/io/harness/accesscontrol/acl/api/ACLResourceImpl.java:81]: We dont need to check if principalToCheckPermissionsFor is notEmpty ?
Also we can write a quick UT when permissionChecksDTOs is empty ? @gupta-utkarsh

Comment (gupta-utkarsh) [reply access-control/service/src/main/java/io/harness/accesscontrol/acl/api/ACLResourceImpl.java]: Principal is handled separately. We don't need to check for that. Added a UT for the same.

**Review by reetika04 — APPROVED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32217)* status: 3 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by manish-manchanda — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32217)* status: :shipit: complete! ✓

**Pull request merged**